### PR TITLE
do not add transport_socket_match to inboud clusters

### PR DESCRIPTION
**Pull request by GregHanson**
Please provide a description for what this PR is for.
if trafficDirection is inbound, explicitly set the tls context to disable else auto mtls kicks in and sets up the transport socket matcher for inbound as well

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by GregHanson** [comment]
it's odd, `clusteropts.direction` is not used anywhere in the code.  also seems odd that in a method called `applyUpstreamTLSSettings` we never check direction anywhere before calling it

File: pilot/pkg/networking/core/v1alpha3/cluster.go [hunk 1]
```diff
@@ -1156,7 +1156,7 @@ func applyUpstreamTLSSettings(env *model.Environment, cluster *apiv2.Cluster, tl
 	}
 
 	// convert to transport socket matcher if the mode was auto detected
-	if tls.Mode == networking.TLSSettings_ISTIO_MUTUAL && mtlsCtxType == autoDetected {
+	if tls.Mode == networking.TLSSettings_ISTIO_MUTUAL && mtlsCtxType == autoDetected && util.IsIstioVersionGE14(proxy) {
```
Comment (incfly) [inline pilot/pkg/networking/core/v1alpha3/cluster.go]: unless you plan to CP this into 1.3 release branch, I don't think this check is needed. right? If you only CP into 1.4

Comment (howardjohn) [reply pilot/pkg/networking/core/v1alpha3/cluster.go]: Pilot 1.4 must work with Envoy 1.3

**Comment by incfly** [comment]
I modified the title, as I understand you want to remove the `transport_socket_matches` not the `Cluster.tls_context` for inbound cluster.

Also this is a clean up of the config, but should not have actual effect in terms of connectivity, right?

File: pilot/pkg/networking/core/v1alpha3/cluster.go (hunk 1 above)
Comment (incfly) [reply pilot/pkg/networking/core/v1alpha3/cluster.go]: Oh I see, thanks for clarifying, I missed that...

**Comment by GregHanson** [comment]
@incfly yes, this is partial config cleanup - as of right not including the transport_socket_matches on the inbound cluster does not produce any failures.  But as John also said, the proxy version check does verify compatibility with older sidecars since they are expecting a cluster.tlscontext to be set in that case

**Comment by howardjohn** [comment]
/retest

test disabled in https://github.com/istio/istio/pull/18053

**Pull request merged**